src/index.ts

```diff
@@ -51,3 +52,11 @@ export const sha512_224: (input: string | Uint8Array) => Uint8Array = hash.sha51
  * @category hashing
  */
 export const sha512_256: (input: string | Uint8Array) => Uint8Array = hash.sha512_256
+
+/**
+ * Password hashing the password and salt with scrypt.
+ *
+ * @since 1.0.0
+ * @category password hashing
+ */
+export const scrypt: typeof passwordHash.scrypt = passwordHash.scrypt
```

Comment:
Let's avoid `typeof` here and "accept" the repetition. I'm also "team inference" but here, the intend is to expose it fully readable for lib consumers.

Alternatively, we can also ditch the `internal/*` modules (until we need them) and just put all the logic from those in-line right here.

Reply:
yeah, let's ditch the internal module. Writing out all the stuff is painful. But is typeof a problem? In VSCode it's showing the correct signature and now the typeof

<img width="836" alt="Screenshot 2024-01-08 at 22 17 36" src="https://github.com/fubhy/effect-crypto/assets/223045/f554eb9a-ef6d-49a7-ba63-2fbab6a4173e">

Reply:
`typeof` is fine. The point here (in the entrypoints of the public modules as well as the root entrypoints) is that having the type signatures written out and paired with the documentation comments  makes these entrypoints extremely readable without having to further traverse into the underlying internals and/or using the IDE. That's all.

Reply:
Ok then, let's ditch the internal modules (whose point really only is to circuit break circular dependencies), until we actually need them.

Reply:
but does this mean you would even copy & paste for example `ScryptOpts` which is internally coming from noble?

My main concern copying the types would be that if e.g. a new optional option gets added with a package upgrade you might miss adding it.

Reply:
For `options` structs no, I'd not repeat them here I think. It's a trade off between expressiveness (DX) and maintenance burden. I personally prefer the DX gain and readability of the lib code (which we also publish through `dist/src`).

Reply:
That sounds like an acceptable trade-off :)